### PR TITLE
Automated cherry pick of #120492: service controller: improve node lifecycle updates - update

### PR DESCRIPTION
**Pull request by cezarygerard**
Cherry pick of #120492 on release-1.28.

#120492: service controller: improve node lifecycle updates - update

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Comment by aojea** [comment]
/lgtm
/approve

/test pull-kubernetes-integration-go-compatibility

Seems unrelated, integration scheduler test

**Comment by code-elinka** [comment]
@andrewsykim @cheftako Could you please take a look?

**Comment by xmudrii** [comment]
@thockin @andrewsykim This cherry-pick is missing approval, other cherry-picks in series are approved

**Comment by thockin** [comment]
@cheftako @liggitt @andrewsykim 

Do we need more approvers for this subdir?

**Comment by liggitt** [comment]
I'm only here for the go.mod file, will defer to Walter and Andrew for actual changes

**Comment by saschagrunert** [comment]
Let's fix the test failures in https://github.com/kubernetes/kubernetes/pull/120945 before approving this one.

**Comment by code-elinka** [comment]
> Let's fix the test failures in #120945 before approving this one.

The changes were fixed in new PRs for 1.26 & 1.27

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
